klippa/src/main.rs

```diff
@@ -0,0 +1,104 @@
+//! binary subset tool
+//!
+//! Takes a font file and a subset input which describes the desired subset, and ouput is a new
+//! font file containing only the data specified in the input.
+//!
+
+use font_types::GlyphId;
+use klippa::{Plan, Subset};
+use read_fonts::{FontRef, TableProvider, TopLevelTable};
+use std::collections::BTreeSet;
+use thiserror::Error;
+use write_fonts::{
+    from_obj::FromTableRef, tables::hhea::Hhea, tables::hmtx::Hmtx, tables::maxp::Maxp, FontBuilder,
+};
+
+#[derive(Error, Debug)]
+pub enum InvalidInputError {
+    #[error("Invalid input gid {0}")]
+    InvalidGidError(String),
+
+    #[error("Invalid gid range {start}-{end}")]
+    InvalidGidRangeError { start: u16, end: u16 },
+}
+
+fn main() {
+    let args = match flags::Args::from_env() {
+        Ok(args) => args,
+        Err(_e) => {
```

Comment:
in both of these branches we're discarding the error, which means we'll just exit silently. We probably want to at least print the error beforehand?